src/frame/mod.rs

```diff
@@ -40,11 +41,24 @@ pub struct Frame {
 impl Frame {
     /// Check if this Frame is identical to another frame
     pub(crate) fn compare(&self, other: &Frame) -> bool {
-        self.id == other.id
-            && self.content.unique() == other.content.unique()
-            && (self.encoding.is_none()
-                || other.encoding.is_none()
-                || self.encoding == other.encoding)
+        if self.id == other.id {
+            let content_eq = if let ID::Valid(id) = &self.id {
+                // some link frames are allowed to have the same id as long their content is different
+                let cmp_deeper = if id == "WCOM" || id == "WOAR" {
```

Comment:
Ok, sorry for taking so long each time, I needed to read up about this in the spec...

It seems this really is a special case for just the WCOM and WOAR frames. So it would be best to also special case the comparison. If we perform the comparison between the content as well, we do not need the `deeper` param to unique so it can be kept to a single branch/match.

Reply:
Could you also introduce a unit test for this new behaviour?

Reply:
So you mean to remove the deeper param and compare the content directly in compare() for WCOM and WOAR?

Reply:
Yes :)